pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -545,16 +545,26 @@ public void shutDownBrokerGracefully(
             @ApiParam(name = "maxConcurrentUnloadPerSec",
                     value = "if the value absent(value=0) means no concurrent limitation.")
             @QueryParam("maxConcurrentUnloadPerSec") int maxConcurrentUnloadPerSec,
-            @QueryParam("forcedTerminateTopic") @DefaultValue("true") boolean forcedTerminateTopic
+            @QueryParam("forcedTerminateTopic") @DefaultValue("true") boolean forcedTerminateTopic,
+            @Suspended final AsyncResponse asyncResponse
     ) {
-        validateSuperUserAccess();
-        doShutDownBrokerGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
+        validateSuperUserAccessAsync()
+                .thenCompose(__ -> doShutDownBrokerGracefullyAsync(maxConcurrentUnloadPerSec, forcedTerminateTopic))
+                .thenAccept(__ -> {
+                    LOG.info("[{}] Successfully shutdown broker gracefully", clientAppId());
+                    asyncResponse.resume(Response.noContent().build());
+                })
```

Comment:
`doShutDownBrokerGracefullyAsync` method will close the web service. After the web service is closed, can `asyncResponse.resume(Response.noContent().build());` work? I think we can resume the response first, then call `doShutDownBrokerGracefullyAsync`. What do you think?

Reply:
I don't think my question is similar to https://github.com/apache/pulsar/pull/20709#discussion_r1252482006. I mean, if we run `doShutDownBrokerGracefullyAsync` first, then run `asyncResponse.resume(Response.noContent().build());`, is that OK? In `doShutDownBrokerGracefullyAsync` method, the admin web service will be shut down, can `asyncResponse.resume` succeed when the web service has shut down?

I'm not sure if my guess is accurate, but I remember encountering a similar situation before. I suggest run `asyncResponse.resume(Response.noContent().build());` first, then run `doShutDownBrokerGracefullyAsync`.